Alluvial.Tests/DataStreamCatchupTests.cs

```diff
@@ -222,21 +222,72 @@ await Catchup.Create(streamSource.Updates())
-                                   {
-                                       WriteEvent(streamId, 1m);
-                                       await Task.Delay(1);
-                                   }
-                               });
+                {
+                    foreach (var streamId in streamIds.Take(200))
+                    {
+                        WriteEvent(streamId, 1m);
+                        await Task.Delay(1);
+                    }
+                });
 
                 await Wait.Until(() =>
-                                 {
-                                     var sum = projectionStore.Sum(b => b.Balance);
-                                     Console.WriteLine("sum is " + sum);
-                                     return sum >= 1200;
-                                 });
+                {
+                    var sum = projectionStore.Sum(b => b.Balance);
+                    Console.WriteLine("sum is " + sum);
+                    return sum >= 1200;
+                });
             }
         }
+
+        [Test]
+        public async Task By_default_a_catchup_run_stops_when_an_aggregator_throws_an_exception()
+        {
+            var projectionStore = new InMemoryProjectionStore<BalanceProjection>();
+
+            var projector = new BalanceProjector()
+                .Pipeline((projection, batch, next) =>
+                {
+                    if (projectionStore.Count() >= 30)
+                    {
+                        throw new Exception("oops");
+                    }
+                    next(projection, batch);
+                });
+
+            var catchup = Catchup.Create(streamSource.Updates(), batchCount: 50)
+                                 .Subscribe(projector, projectionStore);
+
+            Action runSingleBatch = () => catchup.RunSingleBatch().Wait();
+
+            runSingleBatch.ShouldThrow<Exception>()
+                .And
+                .Message
+                .Should()
+                .Contain("oops");
+        }
+
+        [Test]
+        public async Task Catch_allows_aggregator_exceptions_to_be_prevented_from_stopping_catchup()
+        {
+            var projectionStore = new InMemoryProjectionStore<BalanceProjection>();
+
+            var projector = new BalanceProjector()
+                .Pipeline((projection, batch, next) =>
+                {
+                    if (projectionStore.Count() == 30)
+                    {
+                        throw new Exception("oops");
+                    }
+                    next(projection, batch);
+                })
+                .Catch(continueIf: (projection, batch, next) => true);
+
+            var catchup = Catchup.Create(streamSource.Updates(), batchCount: 50)
+                                 .Subscribe(projector, projectionStore);
+
+            await catchup.RunSingleBatch();
+
+            projectionStore.Count().Should().Be(50);

```

Comment:
This is a little unintuitive right now. I expected this initially to be 49, but what happens is that `Catch` prevents the `DataStreamCatchup<T>` from ever seeing the exception, so it updates the projection cursor and moves on. I think these may both be valid scenarios.

I'm leaning toward also putting a catch block inside of `DataStreamCatchup<T>.Aggregate` so that the projection will not be updated. But then should a catchup continuing on error be based on that error handler rather than Catch at the aggregator level?

Thoughts?